src/io/flutter/inspections/WrongModuleTypeNotificationProvider.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2016 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.inspections;
+
+
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.fileEditor.FileEditor;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtilCore;
+import com.intellij.openapi.project.DumbAware;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.ui.Messages;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.ui.EditorNotificationPanel;
+import com.intellij.ui.EditorNotifications;
+import com.intellij.util.containers.ContainerUtil;
+import com.jetbrains.lang.dart.DartFileType;
+import com.jetbrains.lang.dart.util.PubspecYamlUtil;
+import io.flutter.module.FlutterModuleType;
+import io.flutter.sdk.FlutterSdkGlobalLibUtil;
+import io.flutter.sdk.FlutterSdkUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Set;
+
+
+public class WrongModuleTypeNotificationProvider extends EditorNotifications.Provider<EditorNotificationPanel> implements DumbAware {
+  private static final Key<EditorNotificationPanel> KEY = Key.create("Wrong module type");
+  private static final String DONT_ASK_TO_CHANGE_MODULE_TYPE_KEY = "do.not.ask.to.change.module.type";
+  private static final String FLUTTER_YAML_FILE = "flutter.yaml";
+
+  private final Project myProject;
+
+  public WrongModuleTypeNotificationProvider(@NotNull Project project) {
+    myProject = project;
+  }
+
+  private static boolean hasFlutterYaml(Module module) {
+    VirtualFile baseDir = module.getProject().getBaseDir();
+    if (baseDir != null) {
+      VirtualFile flutterYaml = baseDir.findFileByRelativePath(FLUTTER_YAML_FILE);
+      return flutterYaml != null && flutterYaml.exists();
+    }
+    return false;
+  }
+
+  @NotNull
+  private static EditorNotificationPanel createPanel(@NotNull Project project, @NotNull Module module) {
+    EditorNotificationPanel panel = new EditorNotificationPanel();
+    panel.setText("'" + module.getName() + "' is not a Flutter Module, device debugging is not fully supported.");
```

Comment:
I started to and then saw that we're only half doing it.  I'm tempted to not bother since I doubt we'll ever internationalize.

Looking at the Dart plugin it's pretty fastidious.  Go, on the other hand, is not.

Happy to go with whatever you all prefer.